### PR TITLE
diagflat

### LINKED ISSUE #17896: diagflat
Working on (#15024 -> )

### PR DESCRIPTION
**Pull request by Sarvesh-Kesharwani**
Closed #17896

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py [hunk 1]
```diff
@@ -360,3 +360,37 @@ def test_paddle_empty_like(
         x=x[0],
         dtype=dtype[0],
     )
+
+
+# diagflat
+@handle_frontend_test(
+    fn_tree="paddle.diagflat",
+    dtype_and_values=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py:369]: Best to use `helpers.get_dtypes("valid")` here to expand the test coverage

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 2]
```diff
@@ -76,3 +76,8 @@ def eye(num_rows, num_columns=None, dtype=None, name=None):
 @to_ivy_arrays_and_back
 def empty_like(x, dtype=None, name=None):
     return ivy.empty_like(x, dtype=dtype)
+
+
+@to_ivy_arrays_and_back
+def diagflat(x, offset=None, name=None):
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/creation.py:82]: The default value of `offset` is `0` not `None`

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @Sarvesh-Kesharwani, requested some changes. Can you look into those? Thanks! :)

File: ivy/functional/frontends/paddle/tensor/creation.py (hunk 2 above)
Comment (Sarvesh-Kesharwani) [reply ivy/functional/frontends/paddle/tensor/creation.py:82]: Oops!

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 3]
```diff
@@ -805,7 +805,11 @@ def arrays_to_numpy(x):
         rtol = _get_framework_rtol(rtol, ivy.backend)
     if isinstance(atol, dict):
         atol = _get_framework_atol(atol, ivy.backend)
-
+    print("-----------------------------------------------")
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:808]: Please revert those debugging statements, thanks!

**Review by CatB1t — CHANGES_REQUESTED** [review]

**Comment by Sarvesh-Kesharwani** [comment]
chrome_ES5Dwjc6gb

**Comment by Sarvesh-Kesharwani** [comment]
> Hey @Sarvesh-Kesharwani, requested some changes. Can you look into those? Thanks! :)

Hi Haris, i have made the requested changes, please review theme.

**Review by hmahmood24 — APPROVED** [review]
Hey @Sarvesh-Kesharwani. Everything seems good to merge now! 🚀 Thanks!

**Pull request merged**